blogs/ulysses-offload/README.md

```diff
@@ -0,0 +1,263 @@
+# Ulysses-Offload: Democratizing Long Context LLM Training 
+
+<img src="./media/image1.png" style="width:6.5in;height:3.34583in"
+alt="A screenshot of a computer Description automatically generated" />
+
+Figure 1: Ulysses-Offload supports 16x longer sequence lengths at 55% 
+Model FLOPs Utilization (MFU) than NVIDIA Megatron-SP and DeepSpeed Ulysses.
+
+
+To cite and for more technical in depth of this release, please see 
+our [arxiv report](https://arxiv.org/abs/2408.16978):
+
+@article{yao2024ulysses,
+
+title={ Training Ultra Long Context Language Model with Fully Pipelined
+Distributed Transformer},
+
+author={Jinghan Yao and Sam Ade Jacobs and Masahiro Tanaka and Olatunji
+Ruwase and Aamir Shafi and Hari Subramoni and Dhabaleswar K. (DK) Panda
+},
+
+journal={https://arxiv.org/abs/2408.16978},
+
+year={2024}
+
+}
+
+## Introduction
+
+In the rapidly evolving field of generative AI and scientific ML, the
+ability to train large (language) models with ultra-long context
+capabilities is becoming increasingly important. These models are
+essential for a variety of complex tasks, ranging from understanding
+lengthy documents to image and video generation to processing extensive
+sequences in computational biology. However, training such models
+efficiently poses significant challenges due to the enormous GPU
+resources and memory required.
+
+Building on our previous project, DeepSpeed Ulysses, which focuses on
+system optimizations for training extremely long sequence transformer
+models, we are excited to present the Fully Pipelined Distributed
+Transformer (FPDT), also known as Ulysses-Offload, in this release. FPDT
+is an innovative, resource-efficient technique that offers comparable
+benefits to DeepSpeed Ulysses and other previous long-context
+optimization methods, but with a modest hardware budget. FPDT makes
+ultra long-context large language models (LLM) training and finetuning
+accessible to everyone, regardless of GPU resources. FPDT enables
+training with context lengths of up to 2 million tokens using just 4
+NVIDIA A100-40GB GPUs. FPDT supports 16x longer sequence lengths at 55% 
+Model FLOPs Utilization (MFU) than NVIDIA Megatron-SP and DeepSpeed Ulysses
+(see Figure 1). The next section highlights the key innovations of FPDT, 
+and subsequent sections provide additional details on the design and 
+usability of FPDT, followed by experimental results.
+
+## Key Innovations
+
+### 1. Fully Pipelined Distributed Transformer (FPDT)
+
+The core innovation of our work is the Fully Pipelined Distributed
+Transformer (FPDT). This approach leverages a pipelined sequence
+chunking, which allows for the training of LLMs with sequence lengths up
```

Comment:
I find it awkward to read this phrase: `a pipelined sequence chunking`. Is there a way to simplify?